src/security/srv_acl.c

```diff
@@ -538,7 +538,7 @@ filter_cont_capas_based_on_flags(uint64_t flags, uint64_t *capas)
 	if (flags & DAOS_COO_RO)
 		*capas &= CONT_CAPAS_RO_MASK;
 	else if (!(*capas & CONT_CAPAS_RO_MASK) ||
-		 !(*capas & ~CONT_CAPAS_RO_MASK))
+		 !(*capas & CONT_CAPAS_W_MASK))
```

Comment:
The previous logic puzzled me when I was working #11598. Moreover, `(uint64_t)~an_unsigned_32_bit_mask` is bad. This is therefore a good improvement.